scripts/download-realm.js

```diff
@@ -72,7 +72,23 @@ function printProgress(input, totalBytes, archive) {
     });
 }
 
+function copyLocal(archive, destination) {
+    const archivePath = `${process.env.HOME}/${archive}`;
+    const exists = fs.existsSync(archivePath);
+    if (exists) {
+        console.log(`Skipping download for ${archive}. Local archive available at ${archivePath}`);
+        fs.copySync(archivePath, destination);
+        return true;
+    }
+
+    return false;
+}
+
 function download(serverFolder, archive, destination) {
+    if (copyLocal(archive, destination)) {
+        return;
```

Comment:
A lot is happening in this file, but not returning a promise would break the promise-chain in the `acquire` function: https://github.com/realm/realm-js/blob/f6638c322e4f94d0bfb30f7db5b0a5311c50b513/scripts/download-realm.js#L152-L159

Would probably be safer to return `Promise.resolve()` instead of `undefined`.

But isn't this something `shouldSkipAcquire` combined with a lock-file should handle?

Reply:
I was looking into the other PR. yes indeed the result should be a resolved promise. pushed a fix for that

Reply:
shouldSkipAcquire is when you have already downloaded it once. This means that you should really have the required files in the required places to use it. 

this change allows you to have the tgz downloaded manually to skip downloading the tgz for debugging